core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -463,4 +509,290 @@ public void appendTo(PositionsAppender positionsAppender)
             positionsAppender.append(getPositions(), getBlock());
         }
     }
+
+    public static Block createTesVarcharBlock(String... values)
+    {
+        BlockBuilder builder = TEST_VARCHAR.createBlockBuilder(null, 100);
+
+        for (String value : values) {
+            if (value == null) {
+                builder.appendNull();
+            }
+            else {
+                TEST_VARCHAR.writeString(builder, value);
+            }
+        }
+
+        return builder.build();
+    }
+
+    // Fake varchar type to test handling not VariableWidthBlock blocks in SlicePositionAppender
+    static class TestVarcharType
```

Comment:
Why do we need new varchar type? You can use standard varchar with special block type